mrmustard/lab/gates.py

```diff
@@ -499,8 +499,10 @@ def __init__(
         orthogonal_trainable: bool = False,
         modes: Optional[List[int]] = None,
     ):
-        if modes is not None and (num_modes != len(modes)):
-            raise ValueError(f"Invalid number of modes, got {len(modes)} but expected {num_modes}.")
+        if modes is not None and (
+            num_modes != len(modes) or any(mode >= num_modes for mode in modes)
+        ):
+            raise ValueError("Invalid number of modes and the mode list here!")
```

Comment:
I don't understand this error message. It would be useful to make it more verbose for the user.

Reply:
it shouldn't be this, I thought I had changed it. thanks

Reply:
that's odd, I don't have this line in my copy of the branch 🤔